tools/clang/test/HLSLFileCheck/hlsl/types/modifiers/groupshared/groupshared_shadermodels.hlsl

```diff
@@ -115,27 +131,32 @@ PCStruct HSPatch(InputPatch<PosStruct, 3> ip,
 PosStruct HSMain(InputPatch<PosStruct, 3> p,
                  uint ix : SV_OutputControlPointID)
 {
-  output[ix] = getit();
+  output[ix] = foo;
   PosStruct s;
   s.pos = p[ix].pos;
   return s;
 }
 
+// LIBCHK: error: Thread Group Shared Memory not supported from non-compute entry points.
+// LIBCHK: error: Thread Group Shared Memory not supported from non-compute entry points.
+// LIBCHK: error: Thread Group Shared Memory not supported from non-compute entry points.
 // LIBCHK: error: Thread Group Shared Memory not supported from non-compute entry points.
 [shader("domain")]
 [domain("tri")]
 PosStruct DSMain(const OutputPatch<PosStruct, 3> patch,
                  uint ix : SV_PrimitiveID)
 {
-  output[ix] = getit();
+  output[ix] = foo;
   PosStruct v;
   v.pos = patch[0].pos;
   return v;
 }
 
-struct MyPayload {
-  float4 color;
-  uint3 pos;
+
+struct [raypayload] MyPayload {
+  float4 color : write(caller) : read(anyhit,closesthit,miss);
+  float4 color2 : write(miss) : read(caller);
+  uint3 pos : write(caller) : read(anyhit,closesthit,miss);
```

Comment:
These are all the payload access qualifiers required for the newer shader model target